### PR TITLE
Removed unnecessary null pointer checks issue

### PR DESCRIPTION
**Pull request by Luc35b**
Looked through every file to remove unnecessary null pointer checks. Was unsure about change made in first commit in ".../Viewer.cc" so change was reverted in second commit. Signed-off-by: Lucas Baraya <lucasbarayarw@gmail.com>

### CODE REVIEW

File: openvdb/openvdb/tree/LeafBuffer.h [hunk 1]
```diff
@@ -330,7 +330,7 @@ template<typename T, Index Log2Dim>
 inline bool
 LeafBuffer<T, Log2Dim>::deallocate()
 {
-    if (mData != nullptr && !this->isOutOfCore()) {
```
Comment (jmlait) [inline openvdb/openvdb/tree/LeafBuffer.h:333]: This is not the same as we will return false for nullptr and in-core in the previous  code, but now are returning true..... 

I'm not sure of the implications though.

**Review by jmlait — COMMENTED** [review]
Thanks for the PR....  Bad memories of the bad-old days when free(0) crashed :>

File: openvdb/openvdb/tree/LeafBuffer.h (hunk 1 above)
Comment (Luc35b) [reply openvdb/openvdb/tree/LeafBuffer.h]: If the implication is unknown I'll change it back thanks for reviewing.

**Review by kmuseth — APPROVED** [review]
Looks good to me

**Review by jmlait — APPROVED** [review]

**Pull request merged**